.github/workflows/CI.yml

```diff
@@ -160,15 +160,13 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
-    - uses: dtolnay/rust-toolchain@1.56.0
+    - uses: dtolnay/rust-toolchain@1.63.0
```

Comment:
`git grep 1.56` says: Please also update `README.md` and `clippy.toml` (but I guess the later could cause new clippy warnings. Since this PR is already large enough, that's perhaps better done later in a different PR....)

There is also the following, which perhaps should be investigated after the MSRV bump.
```
x11rb-async/src/rust_connection/extensions.rs:    // n.b. notgull: Apparently the resolver for 1.56 is too weak to figure out the lifetimes
x11rb-async/src/rust_connection/extensions.rs:    // n.b. notgull: Apparently the resolver for 1.56 is too weak to figure out the lifetimes
```
I'll try to separate the MSRV bump out to its own PR. If that doesn't help, feel free to ignore my PR (if I get around to creating one).

Reply:
> but I guess the later could cause new clippy warnings. 

Apparently it doesn't. I pushed a branch for the MSRV bump and the CI build succeeded. So I guess now new clippy warnings and you can just incorporate the two missing updates from https://github.com/psychon/x11rb/tree/msrv-bump-1.63 into this PR.

Reply:
Updated the docs